gdal/frmts/wms/wmsmetadataset.cpp

```diff
@@ -662,33 +676,52 @@ void GDALWMSMetaDataset::AddTiledSubDataset(const char* pszTiledGroupName,
 /************************************************************************/
 /*                     AnalyzeGetTileServiceRecurse()                   */
 /************************************************************************/
-
-void GDALWMSMetaDataset::AnalyzeGetTileServiceRecurse(CPLXMLNode* psXML)
+// tiledWMS only
+void GDALWMSMetaDataset::AnalyzeGetTileServiceRecurse(CPLXMLNode* psXML, GDALOpenInfo * poOpenInfo)
 {
+    // Only list tiled groups that contain the string in the open option TiledGroupName, if given
+    char **OpenOptions = poOpenInfo ? poOpenInfo->papszOpenOptions : nullptr;
+    CPLString match(CSLFetchNameValueDef(OpenOptions, "TiledGroupName",""));
+    match.toupper();
+    // Also pass the change patterns, if provided
+    char **changes = CSLFetchNameValueMultiple(OpenOptions, "Change");
+
     CPLXMLNode* psIter = psXML->psChild;
     for(; psIter != nullptr; psIter = psIter->psNext)
     {
         if (psIter->eType == CXT_Element &&
             EQUAL(psIter->pszValue, "TiledGroup"))
         {
-            const char* pszName = CPLGetXMLValue(psIter, "Name", nullptr);
+            CPLString name(CPLGetXMLValue(psIter, "Name", ""));
             const char* pszTitle = CPLGetXMLValue(psIter, "Title", nullptr);
-            if (pszName)
-                AddTiledSubDataset(pszName, pszTitle);
+            if (!name.empty())
+            {
+                if (match.empty())
+                {
+                    AddTiledSubDataset(name, pszTitle, changes);
+                }
+                else {
+                    CPLString NAME(name);
+                    NAME.toupper();
+                    if (std::string::npos != NAME.find(match))
+                        AddTiledSubDataset(name, pszTitle, changes);
+                }
+            }
         }
-        else if (psIter->eType == CXT_Element &&
-            EQUAL(psIter->pszValue, "TiledGroups"))
+        else if (psIter->eType == CXT_Element && EQUAL(psIter->pszValue, "TiledGroups"))
         {
-            AnalyzeGetTileServiceRecurse(psIter);
+            AnalyzeGetTileServiceRecurse(psIter, poOpenInfo);
         }
     }
+    if (changes)
```

Comment:
Redundant.
`CPLFree` and `VSIFree` no-op for `nullptr`

Reply:
Doesn't hurt either and it makes the intention clear.

Reply:
No such intention should ever exist here. It's bad habbit, it's cruft, [incorrect code potential](https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null).

Reply:
Fine, even though I disagree.
I'm not convinced that it's a no-op in all cases, even though in this particular case it doesn't make a difference.

Reply:
Check this out, the default implementation of VSIFree : cpl_vsisimple.cpp line 872
```
if (pData != nullptr)
  free( pData );
```